internal/cmd/cli.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+)
+
+// CLI exposes common dependencies to commands.
+type CLI struct {
+	Stdin  io.Reader
+	Stdout io.Writer
+	Stderr io.Writer
+}
+
+// Output a string to CLI.Stdout. Output is like fmt.Printf except that it always
+// adds a trailing newline.
+// To write output without a trailing newline use CLI.Stdout directly.
+func (c *CLI) Output(format string, args ...interface{}) {
+	fmt.Fprintf(c.Stdout, format+"\n", args...)
+}
+
+// TODO indicates a place where os.Stdout is being used instead of CLI.Stdout.
+// It is temporary, and can be removed once everything is ported to use CLI
+// for output.
+var TODO = os.Stdout
+
+// key is a type to ensure no other package can access the CLI value in context.
+type key struct{}
+
+// ctxKey used to store CLI in the context.
+var ctxKey = key{}
+
+// newCli looks for a CLI stores in context. If one exists, the CLI from
+// context is returned, otherwise a new CLI is created with streams set to the
+// standard input and output streams.
+//
+// newCLI is a shim for testing, allowing tests to use a buffer instead of the
+// standard streams.
+func newCLI(ctx context.Context) *CLI {
+	cli, ok := ctx.Value(ctxKey).(*CLI)
+	if ok {
+		return cli
+	}
+	return &CLI{
+		Stdin:  os.Stdin,
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+}
+
+// PatchCLI returns a context which contains a CLI value with the output streams
+// set to buffers. PatchCLI is used by tests to record the output produced by
+// CLI commands.
+func PatchCLI(ctx context.Context) (context.Context, BufferedStreams) {
```

Comment:
Should this live in a test file where it is only ever used by tests?

Reply:
That would work for now. I think it's likely that we'll end up with more than one package for the CLI commands, which will require this to be in the live code, but we could still find a way to put it into a test-only package.

I'll move it into a test file for now, since we don't have that constraint. We can figure out where to move it when we end up with more packages for CLI commands.